parallelCPUProfiler/ErrorGroupWithRecover recover panic log doesn't contains stack

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
[2023/12/27 21:35:01.575 +08:00] [ERROR] [wait_group_wrapper.go:233] ["panic in error group"] [recover="runtime error: integer divide by zero"]
```
parallelCPUProfiler has similar issue
<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)
contains stack
### 3. What did you see instead (Required)
no
### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/pkg/util/BUILD.bazel
+++ b/pkg/util/BUILD.bazel
@@ -82,6 +82,7 @@ go_test(
         "//pkg/testkit/testsetup",
         "//pkg/types",
         "//pkg/util/fastrand",
+        "//pkg/util/logutil",
         "//pkg/util/memory",
         "@com_github_pingcap_errors//:errors",
         "@com_github_stretchr_testify//assert",

--- a/pkg/util/cpuprofile/cpuprofile.go
+++ b/pkg/util/cpuprofile/cpuprofile.go
@@ -211,7 +211,7 @@ func (p *parallelCPUProfiler) sendToConsumers() {
 	defer func() {
 		p.Unlock()
 		if r := recover(); r != nil {
-			logutil.BgLogger().Error("parallel cpu profiler panic", zap.Any("recover", r))
+			logutil.BgLogger().Error("parallel cpu profiler panic", zap.Any("recover", r), zap.Stack("stack"))
 		}
 	}()
 

--- a/pkg/util/wait_group_wrapper.go
+++ b/pkg/util/wait_group_wrapper.go
@@ -229,8 +229,7 @@ func (g *ErrorGroupWithRecover) Go(fn func() error) {
 	g.Group.Go(func() (err error) {
 		defer func() {
 			if r := recover(); r != nil {
-				// stack is automatically printed
-				logutil.BgLogger().Error("panic in error group", zap.Any("recover", r))
+				logutil.BgLogger().Error("panic in error group", zap.Any("recover", r), zap.Stack("stack"))
 				err = GetRecoverError(r)
 			}
 		}()

--- a/pkg/util/wait_group_wrapper_test.go
+++ b/pkg/util/wait_group_wrapper_test.go
@@ -16,9 +16,12 @@ package util
 
 import (
 	"fmt"
+	"os"
+	"path"
 	"testing"
 	"time"
 
+	"github.com/pingcap/tidb/pkg/util/logutil"
 	"github.com/stretchr/testify/require"
 	"go.uber.org/atomic"
 )
@@ -91,11 +94,33 @@ func TestWaitGroupWrapperCheck(t *testing.T) {
 	require.False(t, wg.check())
 }
 
+func middleF() {
+	var a int
+	_ = 10 / a
+}
+
 func TestNewErrorGroupWithRecover(t *testing.T) {
+	bak := os.Stdout
+	t.Cleanup(func() {
+		os.Stdout = bak
+	})
+	tempDir := t.TempDir()
+	fileName := path.Join(tempDir, "test.log")
+	file, err := os.OpenFile(fileName, os.O_CREATE|os.O_WRONLY, 0644)
+	require.NoError(t, err)
+	os.Stdout = file
+	// InitLogger contains zap.AddStacktrace(zapcore.FatalLevel), so log level
+	// below fatal will not contain stack automatically.
+	require.NoError(t, logutil.InitLogger(&logutil.LogConfig{}))
 	eg := NewErrorGroupWithRecover()
 	eg.Go(func() error {
-		panic("test")
+		middleF()
+		return nil
 	})
-	err := eg.Wait()
-	require.Errorf(t, err, "test")
+	err = eg.Wait()
+	require.ErrorContains(t, err, "runtime error: integer divide by zero")
+	require.NoError(t, file.Close())
+	content, err := os.ReadFile(fileName)
+	require.NoError(t, err)
+	require.Contains(t, string(content), "pkg/util.middleF")
 }

